### ISSUE TITLE
Accessibility Support On Windows Feature Request

### ISSUE THREAD

**Issue by Elshara** [wildstar84/fauxdacious#11]
Hi,
Thank you for continuing to work on this project.
It's been pointed out to me by a visually challenged friend, (who uses a screen reader on Windows) that the current build of Fauxdacious cannot be interacted with using the keyboard cursor. He would love to switch from Foobar 2000, to Fauxdacious full time. As their recent versions of that program disabled old plugin support due to their switch to x64bit based systems. I have not seen Audacious make this move, but he has told me that Audacious proper, is somewhat accessible.
I was wondering if you knew of any reasons why that may be the case. Especially in the department of qt interfaces. Ideally using QT 6, as it has some kind of control landmark rapper put in place to bridge accessible APIs together that do not exist as strongly in older versions.
I tested these settings out using the built in screen reader in Windows, called Narrator.
By default in Audacious, if QT interface is selected, you can press the alt key and hear spoken feedback to enter into menus. This is not the case currently with Fauxdacious.
In the Winamp interface on Windows, it's very different. Using GDK 2.0, no keyboard feedback works to select anything.
I believe my friend is looking for the ability to use keyboard feedback with Fauxdacious. As it exists and is completely accessible using Linux by enabling its built in screen reader, Orca.
Although accessibility is a typical work in progress across all apps, different developers take on unique approaches to address this issue. In an effort to bring more customers relying on audio as a primary method of sense communication to modernize their experience with media as a whole.
I figured I'd let you know about this, so that if you were hoping to gain some feedback about how usable your program is to a wider audience. You'd have their input available to you to think on.
I hope you are well. I currently have switched from VLC to Fauxdacious full time and use it on both Windows and Linux as my primary media player, as the interface has impressed me a great deal.
Brandon has used Foobar2000 since 2007 I believe. He's relied on it for many years, and can still make it work for him. However I wanted to show him the potential of Fauxdacious, and could only do so while using Linux. A choice of operating system he can't switch to yet. But with Fauxdacious, he could expect similar performance on Windows.
I read in your FAQ that you don't like building Windows packages all that much. IT may be unattainable to work on at this time, however your thoughts are welcome.
Thank you for being here for us. Your time is valuable.
Sincerely,
With warm regards,
Elshara

**Comment by wildstar84** [comment]
Thank you for choosing Fauxdacious!  First, let me say that as a programmer having "come of age" long before mice and graphical interfaces became the rage, I find it annoying when I'm unable to navigate a program via the keyboard and have even added some keyboard-navigation ability that Audacious doesn't have (in programs I've written from scratch I've taken care to ensure ability to use everything w/o a mouse), however what I've found is that this seems to be difficult to do in many situations in Audacious/Fauxdacious.  Second, I've never implemented a Qt version on Windows (it's GTK only for now, as I'd have to completely redo my build environment on Windows to add all the Qt stuff - Audacious completely redid theirs when they added Qt - and I've had extra difficulties when I've tried to do that (ie. update some of the dependencies - yes they're old, and I lack both technical skills on Windows and users to justify the effort required and the risk of borking it completely).  You might say that the Windows version is the "neglected stepchild" because I don't use Windows (about the only time I ever log into Windows - I only have an old Windows-7 instance that came on the original hard drive (that normally sits in a drawer) that came with this refurb computer years ago that I only swap it into my computer when I need to build a final Fauxdacious release (So far, you're the first person that's reported to me that they are using it on Windows, so yes, THANK YOU for letting me know!).  This was originally just a "gee whiz, can we build this thing on Windows?" thing, particularly when Audacious rejected a request to create a click-and-run installer for it, so I took it on! ;)   Linux is my daily driver and where my tech. skills reside, so I'm constantly developing for that using the latest Debian testing libraries.  

Unlike Audacious, this is a one-man operation, so what we really need for issues like this would be for someone to step up with some patches / assistance to add this kind of capability for us!  Other examples on my radar are:  the Transiflex (language translations) that Audacious provides that I'm not technically or lynguistically knowledgable about (so they've never been updated since I copied theirs years ago); and my desire to have, but lacking technical knowledge of how to properly create binary deb/rpm packages for Linux (Audacious is broken up into several binary and source packages & I've looked into it and found it was going to be very labor-intensive).  

BTW, I have been very busy lately adding new features and fixing issues, and am gearing up to release a new final release (v4.2-final) which will include a new (GTK) release for Windows soon!  For the moment, I'd focus more on trying to convert and assist your friend to Linux over trying to figure out how to merge the latest Qt-6 or GTK3 into Fauxdacious/Audacious on Windows or see if you/he/she can run the Linux vsn. in Windows' WSL (Linux embedded in Windows that Windows-10 supports). ;-)

Regards,

Jim

**Comment by Elshara** [comment]
Hi,
Thank you for returning a response this quick!
It seems like you're the kind of programmer that takes their time to really build a quality product for both power users and visual users alike. It's a rare breed of computer user now days, but old is gold as I believe to be the case.
Back in the day, when other factors decided how good a computer's experience could be. Making audio sound cards, physical keyboards be clicky or quiet, having your choice of a basic display not take away from the experience of a computer. And ultimately, not having programs use up oodles amount of ram, because they could. Made 32bit Windows and Linux work faster than 64bit versions of the same thing today with better screens.
Many programmers are fed up with the main stream way of doing things, having everything be screen focused or based off of mobile touch phones. I don't call them smart phones, because a smart phone would let you have IO to plug in physical productivity suites. I don't care how nice the screen is, I care that what I use is as fast and efficient as possible. No bloat, no thing you can't disable. But having the option there in case you wanted to have a use case for something. That's all a programmer should ever need to worry about in my opinion. Making a community centered around what they build impressed with something they enjoy looking at or interacting with on equal measure.
Sight isn't everything. Having a visually challenged person as a friend, has taught me this to a great extent. Audio, sight, and touch I believe go hand in hand across all that we do in life. This has always, traditionally, been the case to incorporate all three within cycles. Music is the very embodiment of these cycles, that keeps us connected as a community. But it is the creation of music, any genre, that brings focus to life. I know that without it, we'd be truly lost as a society. To not feel, or wish to grow beyond the confines of our own self interests.
I bring this up because you touched briefly on how I'm the first person to say that I use the Windows version of Fauxdacious. I believe in the future of opportunities across the entire cyberverse. So of course, naturally, I believe in giving products the best chance they can get to live up to their opportunity. Along side which, their creators can enjoy building for the same or similar reasons. To return community spirit in the hands of every member.
I'm glad we agree on the keyboard still playing a crucial role in terms of both its effectiveness, and as a computer user. All my friends are not people who tend to rely on video gaming and mobile phones. Those kinds of relationships never last long, as they are too centered in isolation for moments to carry on in terms of something I can greatly respect. Although gamers can be educational at times, this movement is a developing one. And I find the culture centered around screen memes to be toxic at times. Which is why music has healed my focus in many areas of life.
Often times people forget the value of how computers came to be a thing. Cloud services, renewing subscriptions and needless API updates and changes be damned. Unless it's broke, don't fix it. Only build on what you've created to make it stronger. It's a philosophy that does save lives. Building them up is all apart of the process.
When products aren't centered around monetary gain, they're almost always focused on the users who interact with them. Software has taken a turn for the worse, with all the endless flash that gets put into it. I find that functionality holds the best value, way  more so than flash could ever hope to bling for. All its worth in the end, is a means to an expiry date. When the next design comes out.
You're one of the few programmers out their who build a quality product. I stand behind every update you work on, because it improves the quality of music experience and video experience the way it was meant to be played. Your customization options are second to none. First to none, would be the original producers cut of information that was tweaked until it was ready to be enjoyed. Those are hard to find. But there are places where you can go to get the next best thing, which is lossless. And for that, only a few programs natively support maximizing sampling rates for audio and native resolution for video. Without having to tweak compression ratios to fit false/force industry standards.
I can help you update your Windows if you wish to. As far as I know, there are two places where you could get and install a modern Windows if you had to, or cared to use say, Visual Studio or whatever you prefer to code in.
You probably know about Rufus - USB portable utility for booting custom installations.
https://rufus.ie/en/
It is Windows only, however there are two other tools you could use if you're on Linux and need to boot up a Virtual Machine.
https://www.balena.io/etcher/
https://www.ventoy.net/
Now, to get modern images that won't break anything too much, here are the iso files you need.
https://uupdump.net/
These have helped me a lot when switching operating systems. The reason I personally do not use VMWare is because I believe that if I can do the job for free, I don't need to pay someone to virtualize the experience at the cost of latency and performance. Modern Windows has its own virtualized terminal, as you pointed out. It is a bit of a process to be able to enable Mate Shell so that he can interact with Linux programs using orca, which also has to be built from source. So dual booting is a common occurrence even for non programmers, per say.
I'll see if I can find someone else who'd be willing to contribute more in terms of Windows development for QT6. Git Hub and Freelancer are full of developers who work with these programs. They may be worth looking into, perhaps even copying source code from Audacious Windows may actually fix this issue entirely. They have a Git Hub that is forkable. That way, you can learn about it by analyzing each code segment on your own time if you want.
https://github.com/audacious-media-player/audacious
Hope this helps somewhat. I definitely look forward to 4.2 on Windows. It's amazing, I've found my media center for as long as it will run.
Let me know if there's anything else I can help contribute.
Sincerely,
Elshara

**Comment by wildstar84** [comment]
Thanks for all the kind words!  Yes, I'm very familar w/Audacious's GIT (where do you think I got the code to do my fork?!)

Question for you:  
1) Does Audacious's Qt Windows version work w/Orca / your  friend's access needs?  If not, how is it defecient?  Reason I ask is so I can know if I get Fauxdacious built w/Qt on Windows, would that work?

2) You said you're using Fauxdacious on Linux, so I'm assuming you were able to build it.   Audacious has instructions in the win32 directory in their source / GIT (win32/notes.html) for how to build, and I just looked at them and they now appear to be much simpler than what I had to deal with before (using pacman now to install all the libs vs having to unzip, configure, & build each individually).  Since I don't have an extra Windows installation to try & would have to blow away my ancient (but WORKING) build-tool setup (if I screw this up & I nearly have several times trying to upgrade GTK, etc., there'll likely be no more Fauxd releases for Windows!), one way you could help would be if you could try building Fauxdacious on Windows following Audacious's instructions (plus installing libsdl2, libdvdread & libdvdnav - for our video / DVD plugins) & carefully document & let me know any issues you run into, and, if successful, I'll then give it a try here.  Also, take a glance at my Windows build-notes (same place in my GIT but older & include the extra libs I mentioned).  If you don't mind taking a stab at it, it'd be a huge help & I and Fauxdacious would be much obliged!!  Even if you don't fully get Fauxd working that way, I'm mainly interested in any issues installing & setting up the build environment!

Now as far as Qt-6 goes, if you need that for Orca, etc.  you may be in for a long wait as Audacious hasn't even upgraded to Qt-6, though they've been prepping for it (and we've been merging their patches, and except for features / tweaks we add, we pretty much follow them).  I'm not sure Qt-6 has even been released as open-source yet (The Qt Company prefers to release closed-source / commercial first, then wait at least a year before releasing open-source).

**Comment by wildstar84** [comment]
Oh, I do have one more (curiosity) question for you - How did you hear a/b Fauxdacious?

**Comment by wildstar84** [comment]
Closing due to lack of user-response to my questions!
